.github/workflows/publish-test.yml

```diff
@@ -1,30 +1,36 @@
 name: Publish Supervision Pre-Releases to PyPI and TestPyPI
+
 on:
   push:
     tags:
       - "[0-9]+.[0-9]+[0-9]+.[0-9]+a[0-9]"
       - "[0-9]+.[0-9]+[0-9]+.[0-9]+b[0-9]"
       - "[0-9]+.[0-9]+[0-9]+.[0-9]+rc[0-9]"
-
   workflow_dispatch:
 
+# Restrict permissions by default
+permissions: {}
+
 jobs:
   build-and-publish-pre-release-pypi:
     name: Build and publish to PyPI
     runs-on: ubuntu-latest
     environment: test
+    timeout-minutes: 15
     permissions:
-      id-token: write
+      id-token: write  # Required for PyPI publishing
+      contents: read   # Required for checkout
```

Comment:
Out of curiousity, is this explicitly clarifying the permissions level we already had? Or were we granting `write` accidentally?

Reply:
Its better to explicitly clarify permission levels; the default permission levels depend on the GITHUB_TOKEN being used (need to check that in the repo). More info here: https://graphite.dev/guides/github-actions-permissions